csharp/example/Program.cs

```diff
@@ -0,0 +1,134 @@
+﻿/*
+ * Copyright (C) 2023 Dremio Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using Microsoft.Data.Analysis;
+using Grpc.Net.Client;
+using Grpc.Core;
+using Apache.Arrow.Flight.Client;
+using Apache.Arrow.Flight;
+using Apache.Arrow;
+
+namespace FlightClientExample
+{
+    public class Program
+    {
+        public static async Task Main(string[] args)
+        {
+            string host = "localhost";
+            string port = "32010";
+            string user = "dremio";
+            string pass = "dremio123";
+            // For default query, add the Samples source and Format the zips.json in Dremio UI
+            string query = "SELECT city, loc[0] AS longitude, loc[1] AS latitude, pop, state, _id FROM Samples.\"samples.dremio.com\".\"zips.json\" LIMIT 100";
+            string protocol = "http";
+
+            // Parse command-line arguments to override the defaults when set
+            for (int i = 0; i < args.Length-1; i++)
+            {
+                if (i % 2 == 0 && args[i].StartsWith("-")) 
+                {
+                    var key = args[i];
+                    var value = args[i+1];
+                    if (key.EndsWith("host")) host = value;
+                    if (key.EndsWith("port")) port = value;
+                    if (key.EndsWith("user")) user = value;
+                    if (key.EndsWith("pass")) pass = value;
+                    if (key.EndsWith("query")) query = value;
+                    if (key.EndsWith("protocol")) protocol = value;
+                }
+            }
+
+            // Basic auth using username and password

```

Comment:
Creating a separate method for this would modularise and make the code more legible.